src/NHibernate/SqlCommand/SqlString.cs

```diff
@@ -1024,6 +1024,19 @@ public SqlString GetSubselectString()
 			return new SubselectClauseExtractor(this).GetSqlString();
 		}
 
+		internal void SubstituteBogusParameters(IReadOnlyList<Parameter> actualParams, Parameter bogusParam)
+		{
+			int index = 0;
+			var keys = _parameters.Keys;
+			// ReSharper disable once ForCanBeConvertedToForeach
+			for (var i = 0; i < keys.Count; i++)
```

Comment:
Why not using a `foreach` here?

Reply:
Because of modifications in collection

Reply:
The code changes the value associated to the key, but it does not change the keys list, since it is only affecting existing keys.
Unfortunately the `SortedList` implementation, though not changing the underlying keys collection at all in this case, has a single "version" check logic for throwing on all iterations in case of any kind of modification.
So ok.